### PR TITLE
[Feature] Filter by user on edition only - use in edition

### PR DESCRIPTION
**Pull request by rldhont**
## Description

Last part of the new feature, restrict filter by user on edition only https://github.com/3liz/lizmap-web-client/pull/1775, based on lizmap plugin config 3liz/lizmap-plugin#265

Using the filter by login in edition.

Funded by Conseil Départmental du Calvados

### CODE REVIEW

**Comment by nboisteault** [comment]
@rldhont can you give a scenario to test this feature?

**Comment by rldhont** [comment]
> @rldhont can you give a scenario to test this feature?

* Create QGIS project with a PostGIS layer
* Config the project for QGIS server
* Published the PostGIS layer in WFS
* In lizmap plugin, define the PostGIS Layer as editable
* In lizmap plugin, define a filter by login for the PostGIS Layer with the option only for edition https://github.com/3liz/lizmap-plugin/pull/265
* Published the project to lizmap
* Try to edit features with different users in different groups

File: lizmap/modules/lizmap/classes/qgisExpressionUtils.class.php [hunk 1]
```diff
@@ -89,28 +97,54 @@ public static function updateExpressionByUser($layer, $expression)
         }
```
Comment (laurentj) [inline lizmap/modules/lizmap/classes/qgisExpressionUtils.class.php:97]: on the line above, it returns $expression, which does not exist anymore

**Review by laurentj — CHANGES_REQUESTED** [review]

**Comment by nboisteault** [comment]
@rldhont is it ready for tests?

**Comment by nboisteault** [comment]
@rldhont I can't edit this feature while I'm connected as admin user.
Also edition and delete buttons are visible even if user doesn't have rights to do so. I think, this might handled client side.
image

**Comment by rldhont** [comment]
@nboisteault do you have some logs ?

File: lizmap/modules/lizmap/classes/qgisExpressionUtils.class.php (hunk 1 above)
Comment (rldhont) [reply lizmap/modules/lizmap/classes/qgisExpressionUtils.class.php:97]: fixed

**Review by nboisteault — APPROVED** [review]
I have tested. Behavior is ok.

**Pull request merged**